projects/igniteui-angular/src/lib/grid/grid.component.ts

```diff
@@ -323,6 +323,9 @@ export class IgxGridComponent implements OnInit, OnDestroy, AfterContentInit, Af
         this._columnHiding = value;
```

Comment:
Please add a check whether the new value is different than the current value and apply the new value only if there is need.